includes/abilities-api/class-wp-abilities-registry.php

```diff
@@ -98,6 +99,24 @@ public function register( string $name, array $args ): ?WP_Ability {
 		 */
 		$args = apply_filters( 'register_ability_args', $args, $name );
 
+		// Validate category exists if provided (will be validated as required in WP_Ability).
+		if ( isset( $args['category'] ) ) {
+			$category_registry = WP_Abilities_Category_Registry::get_instance();
+			if ( ! $category_registry->is_registered( $args['category'] ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
+						/* translators: %1$s: category slug, %2$s: ability name */
+						esc_html__( 'Category "%1$s" is not registered. Please register the category before assigning it to ability "%2$s".' ),
```

Comment:
My consistent minor message about verbiage. Although here might be less necessary as the message is quite detailed:
```suggestion
						/* translators: %1$s: ability category slug, %2$s: ability name */
						esc_html__( 'Ability category "%1$s" is not registered. Please register the category before assigning it to ability "%2$s".' ),
```